sources/packages/backend/apps/api/src/route-controllers/designation-agreement/models/designation-agreement.dto.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  DesignationAgreementStatus,
+  NOTE_DESCRIPTION_MAX_LENGTH,
+} from "@sims/sims-db";
+import { Type } from "class-transformer";
+import {
+  IsArray,
+  IsDate,
+  IsEnum,
+  IsNotEmpty,
+  IsOptional,
+  MaxLength,
+  ValidateNested,
+} from "class-validator";
+import { UpdateDesignationLocation } from "./designation-agreement.model";
+
+/**
+ * Approve/Deny a designation agreement.
+ * startDate, endDate and locationsDesignations used only for approval.
+ */
+export class UpdateDesignationAPIInDto {
+  @IsEnum(DesignationAgreementStatus)
+  designationStatus: DesignationAgreementStatus;
+  @IsOptional()
```

Comment:
Since "startDate, endDate and locationsDesignations" are required for approval, instead of `@IsOptional()` these fields should be decorated with a `@ValidateIf`+`IsRequired` that will check if the `designationStatus` is "approved".